File: LLD-Python/OOP_Fundamentals/Class/car.py
class Car:
    def __init__(self, brand, model):
        self._brand = brand
        self._model = model
        self._speed = 0

    # Method to acclerate
    def accelerate(self, increment):
        self._speed += increment

    # Method to display info
    def display_status(self):
        print(f"{self._brand} is running at {self._speed} km/h.")
        

File: LLD - Golang/OOP/class/car.go
package main

import "fmt"

// Car struct (similar to class)
type Car struct {
	brand string // unexpected (private to package)
	model string
	speed int
}

// Constructor-like function
func NewCar(brand, model string) *Car {
	return &Car{
		brand: brand,
		model: model,
		speed: 0,
	}
}

// Method to accelerate
func (c *Car) Accelerate(increment int) {
	c.speed += increment
}

// Method to display status
func (c *Car) DisplayStatus() {
	fmt.Printf("%s is running at %d km/h.\n", c.brand, c.speed)
}

// func main() {
// 	car := NewCar("BMW", "X5")
// 	car.Accelerate(50)
// 	car.DisplayStatus()
// }
